### PR TITLE
Feature/dt 1270 mapping update for ncdhc

### PR DESCRIPTION
**Pull request by mdellabitta**
- A bunch of PEP8 fixes.
- Extra conditional on line 149 to suppress error logs
- Removed validate_mapv3 enrichment for collections
- Added intermediateProvider mapping in lines 181 - 201 (piggybacks on the map_data_provider method).

### CODE REVIEW

File: lib/mappers/digitalnc_mapper.py [hunk 1]
```diff
@@ -178,19 +180,25 @@ def map_is_part_of(self):
...
-        _dict = {"dataprovider": None}
+        data_providers = []
 
         if exists(self.provider_data, prop):
             for s in iterify(getprop(self.provider_data, prop)):
                 try:
                     if s.get("type") == "ownership":
-                        _dict["dataProvider"] = s.get("#text")
-                        break
+                        data_providers.append(s.get("#text"))
```
Comment (markbreedlove) [inline lib/mappers/digitalnc_mapper.py]: You can use the `textnode()` method that's in the imports to safely get the text. See elsewhere where it's used.

**Comment by markbreedlove** [comment]
This looks pretty clear. Do you want to squash the commits?

**Comment by markbreedlove** [comment]
Er, sorry. I just had that one comment about `textnode`. Maybe after that's changed?

**Comment by mdellabitta** [comment]
@markbreedlove Is that what you meant?

**Comment by markbreedlove** [comment]
Yes, that's the thing.

**Pull request merged**